internal/sysfs/poll.go

```diff
@@ -0,0 +1,35 @@
+package sysfs
+
+import (
+	"time"
+
+	"github.com/tetratelabs/wazero/experimental/sys"
+)
+
+// Poll exposes the poll(2) syscall.
+//
+// # Notes on Parameters
+//
+//	We currently expose a pointer to a time.Duration instead of a plain integer.
+//	A nil pointer means "wait forever" whereas poll(2) uses a negative value instead.
+//	This may change in the future.
+//
+// # Notes on the Syscall
+//
+//	Because this is a blocking syscall, it will also block the carrier thread of the goroutine,
+//	preventing any means to support context cancellation directly.
+//
+//	There are ways to obviate this issue. We outline here one idea, that is however not currently implemented.
+//	A common approach to support context cancellation is to add a signal file descriptor to the set,
+//	e.g. the read-end of a pipe or an eventfd on Linux.
+//	When the context is canceled, we may unblock a Select call by writing to the fd, causing it to return immediately.
+//	This however requires to do a bit of housekeeping to hide the "special" FD from the end-user.
+func Poll(fds []PollFd, timeout *time.Duration) (int, sys.Errno) {
```

Comment:
Did you consider using a negative value as "no timeout" instead of a pointer?

Reply:
TBH I just used the duration ptr only because select was like that. Since the underlying call uses negative values that way, I guess it's doable

Reply:
that's using non pointer would be worth a shot since that way we could reduce allocation

Reply:
I'm not sure how I messed this up as certainly the original API should used normal numeric type for duration, as that maps to the underlying syscall. I'll fix PollRead in another PR